src/sentry/tagstore/base.py

```diff
@@ -435,3 +435,16 @@ def update_group_tag_key_values_seen(self, project_id, group_ids):
         >>> update_group_tag_key_values_seen(1, [2, 3])
         """
         raise NotImplementedError
+
+    def get_group_tag_keys_and_top_values(self, project_id, group_id, environment_id, user=None):
+        from sentry.api.serializers import serialize
+
+        tag_keys = self.get_group_tag_keys(project_id, group_id, environment_id)
+
+        return [dict(
+            totalValues=self.get_group_tag_value_count(
+                project_id, group_id, environment_id, tk.key),
+            topValues=serialize(self.get_top_group_tag_values(
+                project_id, group_id, environment_id, tk.key, limit=10)),
+            **serialize([tk])[0]
```

Comment:
This delegates the serialization of `{id: , tag:, name:, etc.}` to the registered serializer for `GroupTagKey` rather than reimplementing it.